tests/global_args.rs

```diff
@@ -29,3 +29,86 @@ fn issue_1076() {
     let _ = app.try_get_matches_from_mut(vec!["myprog"]);
     let _ = app.try_get_matches_from_mut(vec!["myprog"]);
 }
+
+#[test]
+fn propagate_global_arg_in_subcommand_to_subsubcommand_1385() {
+    let m1 = App::new("foo")
+        .subcommand(
+            App::new("sub1")
+                .arg(Arg::new("arg1").long("arg1").takes_value(true).global(true))
+                .subcommand(App::new("sub1a")),
+        )
+        .get_matches_from(&["foo", "sub1", "--arg1", "v1", "sub1a"]);
+    assert_eq!(
+        "v1",
+        m1.subcommand_matches("sub1")
+            .unwrap()
+            .subcommand_matches("sub1a")
+            .unwrap()
+            .value_of("arg1")
+            .unwrap()
+    );
+}
+
+#[test]
+fn propagate_global_arg_in_subcommand_to_subsubcommand_2053() {
+    #[derive(Clap, Debug, PartialEq)]
+    struct Opts {
```

Comment:
No, don't use the derive macro, but use the `clap::App` builder. As I said, the bug is not derive specific.

Reply:
But this is what the issue 2053 provides us......

Reply:
Yes, but the bug would still exist if the code is converted to normal builder API. The bug is not in the derive expansion, and thus we should not use the derives to test it.

Reply:
Get it.